### PR TITLE
#3822: Added contentOffset to MediaHttpUploader

### PR DESCRIPTION
**Pull request by andrey-qlogic**
and AbstractGoogleClientRequest.

Fixes https://github.com/googleapis/google-cloud-java/issues/3822 (it's a good idea to open an issue first for discussion)

- [ ] Tests pass
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by sduskis** [comment]
@andrey-qlogic signed the CLA, but didn't comment with "I signed it!"

**Comment by andrey-qlogic** [comment]
I signed it!

**Comment by sduskis** [comment]
@andrey-qlogic, does this fix an issue?  If so can, you please add the issue `#` in the description?

File: google-api-client/src/main/java/com/google/api/client/googleapis/media/MediaHttpUploader.java [hunk 1]
```diff
@@ -351,6 +360,7 @@ private HttpResponse directUpload(GenericUrl initiationRequestUrl) throws IOExce
       initiationRequestUrl.put("uploadType", "multipart");
     } else {
       initiationRequestUrl.put("uploadType", "media");
+      initiationRequestUrl.put("contentOffset", String.valueOf(contentOffset));
```
Comment (chingor13) [inline google-api-client/src/main/java/com/google/api/client/googleapis/media/MediaHttpUploader.java]: Should this be set to `-1` if `contentOffset` is not set by the user? Seems like this is new, unexpected behavior unless there's a specification for uploads saying to contentOffset of -1 is normal.

**Comment by andrey-qlogic** [comment]
@chingor13 thanks for review, the good point about -1 value.

**Comment by andrey-qlogic** [comment]
@chingor13 The comment fixed: Don't add contentOffset when it -1

**Comment by andrey-qlogic** [comment]
Closing this as googleapis/google-cloud-java/issues/3822 was fixed by googleapis/google-cloud-java/pull/4170

**Pull request closed without merging**